### DISCUSSION TITLE
Boot Loop Investigation

### DISCUSSION THREAD

**Discussion by InstantCassette** [alienatedsec/yi-hack-v5#291] [Q&A] [answered]
One camera went into boot loop after a few days running 0.3.8 -> 0.4.0.
Yesterday another one also went into boot loop after router went down for a while.
What's up ? And how to I fix them ?

**Comment by alienatedsec** [comment, marked as answer]
@Zerreth Apologies that was the case for you - I also had my 1080p model doing the same recently. Which model cameras are yours?

The only way to recover was to re-flash with any previous version e.g., from any available [here](https://app.box.com/s/cibs7n1mgvhqaqjlidtveegu1uajt5yr/folder/23494084746) and flashing back to baseline `0.3.8`. You can leave the `yi-hack-v5` folder untouched as the baseline files are the most important.

So far, there are no camera storage problems, either rootfs or home partitions are okay, and when I went through the logs, my camera was indicating the ARP check failure. It has been working for a few days since, and I am monitoring it.

**Comment by InstantCassette** [comment]
Model: YYS.2016, so a 4 pack Home 1080p (48US). I put the `rootfs_y20` & `home_y20` from your link on the sd root and it flashed. Then after the camera booted, replaced it with the `rootfs_y20` & `home_y20` from 0.3.8 and now they're both back. Thanks !

**Comment by InstantCassette** [comment]
I just had it happen again to one of my other working cameras when updating from 0.4.0 to 0.4.1a prerelease. So that's 3 out of 4 cameras.  😅 (update 5 Sept 2023, now 4/4 cameras!)

**Reply by alienatedsec** [reply]
@Zerreth have you used earlier v5 hack versions on those cameras? If so, there could be some artefact that impacts the latest baseline version. Mine is still working fine since reflashing and I hope it will stay this way.

**Reply by InstantCassette** [reply]
Yeah those cameras all have been through prety much a lot of yi hack v4 and v5.

**Comment by Porkenstien** [comment]
Howdy,
@alienatedsec  and @Zerreth I think this issue should remain open as my Yi Outdoor went into a bootloop.
Camera was online on Monday(Oct 9,2023) but when I went to check for updates on Wednesday(Oct 11, 2023) the camera was in a bootloop and no longer accessible.
FYI: This camera had been up since 0.4.1a without being restarted. One of my servers monitors power outages, but doesn't show any blips here since May 2023.
Will try to follow previous instructions to get it back in a running state, will post as needed.
Thanks,
Joe

**Reply by alienatedsec** [reply]
@Porkenstien @Zerreth 

I am unsure of the root cause for the boot loop, but I know what executes it.
- The camera is starting and tries to connect to a WiFi network, but is unsuccessful - reason unknown
- The "watchdog" in the camera firmware (not a hack) is rebooting automatically if no WiFi for a certain amount of time

My `yi_dome` model had this recently, and I turned it off and left it unplugged for several days. I turned it back on and it connected. I rebooted the camera again, and it was back in the loop. Finally, I reflashed it as per the [above](https://github.com/alienatedsec/yi-hack-v5/discussions/291#discussioncomment-6837396) and it has worked since.

I have several WiFi networks around my house, and a couple of routers with the same ESSID (for different reasons), so if your setup is similar, this could be the root cause.

**Reply by InstantCassette** [reply]
@alienatedsec  Reflashing will only temporarily fix it. After x-amount of time they will eventually all end up in a boot loop again. (All my 4x Yi Home 1080p (48US) go into boot loops eventually.) 
My setup is 2x Ubiquiti U6-Enterprise, and they send out the same SSID for each respective band. (2SPECTRUM for 2.4Ghz, 5SPECTRUM for 5Ghz & 6SPECTRUM for 6Ghz).

Pure speculation: Could it be that they try to connect to the other AP than the one it was setup with and it freaks out that the MAC or whatever doesn't match ? And just simply fails, reboot & try again ? Is there any way to bypass this behaviour ?

**Reply by alienatedsec** [reply]
> Reflashing will only temporarily fix it.

@Zerreth you may be 100% right here as it happened across all of my devices at least twice to date.

> Could it be that they try to connect to the other AP than the one it was setup with and it freaks out that the MAC or whatever doesn't match ? And just simply fails, reboot & try again ?

I had the same thoughts and the only way to test is to reset the camera, have only one AP with a unique ESSID, and pair the camera with that WiFi network. Finally, leave it running for several weeks.

> Is there any way to bypass this behaviour?

This generation of YI cameras is limited by the baseline firmware supplied by YI and its storage; hence, I am not surprised the maintenance stopped with earlier hack versions. Working on this `yi-hack-v5` repo made me realise this as well, and that's why we have separate hacks on the camera (the hack baseline) and the SD (`yi-hack-v5` folder).

If there is anyone with the knowledge and would like to contribute to how `/home/base/tools/wpa_supplicant` works or could be amended, then I am happy to accept any PRs. It feels like we will need an updated cross-compile of the latest version to better support modern WiFi networks.

I currently have limited time to investigate it promptly.

**Reply by InstantCassette** [reply]
> I had the same thoughts and the only way to test is to reset the camera, have only one AP with a unique ESSID, and pair the camera with that WiFi network. Finally, leave it running for several weeks.

Reset all 4 of my cameras, setup on a new 2.4Ghz SSID broadcasting from a single AP. Now I'll wait and see. If this ends in failure i'm likely going to be dumping these and just shell out for a dahua or hikvision setup.

**Reply by alienatedsec** [reply]
@Zerreth the same crossed my mind many times and if not this repo, I would abandon those cameras a long time ago. However, I will work on a new `wpa_suppliant` version to replace the one supplied with the YI firmware.

Also, it's worth checking the preamble settings as those are important for older Wi-Fi setups https://routerguide.net/preamble-type-short-or-long/

**Reply by InstantCassette** [reply]
> @Zerreth the same crossed my mind many times and if not this repo, I would abandon those cameras a long time ago. However, I will work on a new `wpa_suppliant` version to replace the one supplied with the YI firmware.

2 out of 4 cameras have already started boot looping again. 1st one was after router went down. 2nd one was after unplugging the camera to move it somewhere else.

**Reply by alienatedsec** [reply]
The `wpa_supplicant` version on the cam is `v2.6`

```
wpa_supplicant v2.6
Copyright (c) 2003-2016, Jouni Malinen <j@w1.fi> and contributors
/home/yi-hack-v5 #
```

When compared to the release graph it is 7 years old - https://w1.fi/releases.html

I will try my best to fix it with a new release, but I haven't found a way to compile it as of yet. Looking at the supported versions, it does not mention WPA3, so mixed settings where WPA3 is present is not recommended either.

```
Supported WPA/IEEE 802.11i features
WPA-PSK ("WPA-Personal")
WPA with EAP (e.g., with RADIUS authentication server) ("WPA-Enterprise")
key management for CCMP, TKIP, WEP104, WEP40
WPA and full IEEE 802.11i/RSN/WPA2
RSN: PMKSA caching, pre-authentication
IEEE 802.11r
IEEE 802.11w
Wi-Fi Protected Setup (WPS)
```

**Reply by alienatedsec** [reply]
Changelog

```
2022-01-16 - v2.10
	* SAE changes
	  - improved protection against side channel attacks
	    [https://w1.fi/security/2022-1/]
	  - added support for the hash-to-element mechanism (sae_pwe=1 or
	    sae_pwe=2); this is currently disabled by default, but will likely
	    get enabled by default in the future
	  - fixed PMKSA caching with OKC
	  - added support for SAE-PK
	* EAP-pwd changes
	  - improved protection against side channel attacks
	  [https://w1.fi/security/2022-1/]
	* fixed P2P provision discovery processing of a specially constructed
	  invalid frame
	  [https://w1.fi/security/2021-1/]
	* fixed P2P group information processing of a specially constructed
	  invalid frame
	  [https://w1.fi/security/2020-2/]
	* fixed PMF disconnection protection bypass in AP mode
	  [https://w1.fi/security/2019-7/]
	* added support for using OpenSSL 3.0
	* increased the maximum number of EAP message exchanges (mainly to
	  support cases with very large certificates)
	* fixed various issues in experimental support for EAP-TEAP peer
	* added support for DPP release 2 (Wi-Fi Device Provisioning Protocol)
	* a number of MKA/MACsec fixes and extensions
	* added support for SAE (WPA3-Personal) AP mode configuration
	* added P2P support for EDMG (IEEE 802.11ay) channels
	* fixed EAP-FAST peer with TLS GCM/CCM ciphers
	* improved throughput estimation and BSS selection
	* dropped support for libnl 1.1
	* added support for nl80211 control port for EAPOL frame TX/RX
	* fixed OWE key derivation with groups 20 and 21; this breaks backwards
	  compatibility for these groups while the default group 19 remains
	  backwards compatible
	* added support for Beacon protection
	* added support for Extended Key ID for pairwise keys
	* removed WEP support from the default build (CONFIG_WEP=y can be used
	  to enable it, if really needed)
	* added a build option to remove TKIP support (CONFIG_NO_TKIP=y)
	* added support for Transition Disable mechanism to allow the AP to
	  automatically disable transition mode to improve security
	* extended D-Bus interface
	* added support for PASN
	* added a file-based backend for external password storage to allow
	  secret information to be moved away from the main configuration file
	  without requiring external tools
	* added EAP-TLS peer support for TLS 1.3 (disabled by default for now)
	* added support for SCS, MSCS, DSCP policy
	* changed driver interface selection to default to automatic fallback
	  to other compiled in options
	* a large number of other fixes, cleanup, and extensions

2019-08-07 - v2.9
	* SAE changes
	  - disable use of groups using Brainpool curves
	  - improved protection against side channel attacks
	  [https://w1.fi/security/2019-6/]
	* EAP-pwd changes
	  - disable use of groups using Brainpool curves
	  - allow the set of groups to be configured (eap_pwd_groups)
	  - improved protection against side channel attacks
	  [https://w1.fi/security/2019-6/]
	* fixed FT-EAP initial mobility domain association using PMKSA caching
	  (disabled by default for backwards compatibility; can be enabled
	  with ft_eap_pmksa_caching=1)
	* fixed a regression in OpenSSL 1.1+ engine loading
	* added validation of RSNE in (Re)Association Response frames
	* fixed DPP bootstrapping URI parser of channel list
	* extended EAP-SIM/AKA fast re-authentication to allow use with FILS
	* extended ca_cert_blob to support PEM format
	* improved robustness of P2P Action frame scheduling
	* added support for EAP-SIM/AKA using anonymous@realm identity
	* fixed Hotspot 2.0 credential selection based on roaming consortium
	  to ignore credentials without a specific EAP method
	* added experimental support for EAP-TEAP peer (RFC 7170)
	* added experimental support for EAP-TLS peer with TLS v1.3
	* fixed a regression in WMM parameter configuration for a TDLS peer
	* fixed a regression in operation with drivers that offload 802.1X
	  4-way handshake
	* fixed an ECDH operation corner case with OpenSSL

2019-04-21 - v2.8
	* SAE changes
	  - added support for SAE Password Identifier
	  - changed default configuration to enable only groups 19, 20, 21
	    (i.e., disable groups 25 and 26) and disable all unsuitable groups
	    completely based on REVmd changes
	  - do not regenerate PWE unnecessarily when the AP uses the
	    anti-clogging token mechanisms
	  - fixed some association cases where both SAE and FT-SAE were enabled
	    on both the station and the selected AP
	  - started to prefer FT-SAE over SAE AKM if both are enabled
	  - started to prefer FT-SAE over FT-PSK if both are enabled
	  - fixed FT-SAE when SAE PMKSA caching is used
	  - reject use of unsuitable groups based on new implementation guidance
	    in REVmd (allow only FFC groups with prime >= 3072 bits and ECC
	    groups with prime >= 256)
	  - minimize timing and memory use differences in PWE derivation
	    [https://w1.fi/security/2019-1/] (CVE-2019-9494)
	* EAP-pwd changes
	  - minimize timing and memory use differences in PWE derivation
	    [https://w1.fi/security/2019-2/] (CVE-2019-9495)
	  - verify server scalar/element
	    [https://w1.fi/security/2019-4/] (CVE-2019-9499)
	  - fix message reassembly issue with unexpected fragment
	    [https://w1.fi/security/2019-5/]
	  - enforce rand,mask generation rules more strictly
	  - fix a memory leak in PWE derivation
	  - disallow ECC groups with a prime under 256 bits (groups 25, 26, and
	    27)
	* fixed CONFIG_IEEE80211R=y (FT) build without CONFIG_FILS=y
	* Hotspot 2.0 changes
	  - do not indicate release number that is higher than the one
	    AP supports
	  - added support for release number 3
	  - enable PMF automatically for network profiles created from
	    credentials
	* fixed OWE network profile saving
	* fixed DPP network profile saving
	* added support for RSN operating channel validation
	  (CONFIG_OCV=y and network profile parameter ocv=1)
	* added Multi-AP backhaul STA support
	* fixed build with LibreSSL
	* number of MKA/MACsec fixes and extensions
	* extended domain_match and domain_suffix_match to allow list of values
	* fixed dNSName matching in domain_match and domain_suffix_match when
	  using wolfSSL
	* started to prefer FT-EAP-SHA384 over WPA-EAP-SUITE-B-192 AKM if both
	  are enabled
	* extended nl80211 Connect and external authentication to support
	  SAE, FT-SAE, FT-EAP-SHA384
	* fixed KEK2 derivation for FILS+FT
	* extended client_cert file to allow loading of a chain of PEM
	  encoded certificates
	* extended beacon reporting functionality
	* extended D-Bus interface with number of new properties
	* fixed a regression in FT-over-DS with mac80211-based drivers
	* OpenSSL: allow systemwide policies to be overridden
	* extended driver flags indication for separate 802.1X and PSK
	  4-way handshake offload capability
	* added support for random P2P Device/Interface Address use
	* extended PEAP to derive EMSK to enable use with ERP/FILS
	* extended WPS to allow SAE configuration to be added automatically
	  for PSK (wps_cred_add_sae=1)
	* removed support for the old D-Bus interface (CONFIG_CTRL_IFACE_DBUS)
	* extended domain_match and domain_suffix_match to allow list of values
	* added a RSN workaround for misbehaving PMF APs that advertise
	  IGTK/BIP KeyID using incorrect byte order
	* fixed PTK rekeying with FILS and FT

2018-12-02 - v2.7
	* fixed WPA packet number reuse with replayed messages and key
	  reinstallation
	  [https://w1.fi/security/2017-1/] (CVE-2017-13077, CVE-2017-13078,
	  CVE-2017-13079, CVE-2017-13080, CVE-2017-13081, CVE-2017-13082,
	  CVE-2017-13086, CVE-2017-13087, CVE-2017-13088)
	* fixed unauthenticated EAPOL-Key decryption in wpa_supplicant
	  [https://w1.fi/security/2018-1/] (CVE-2018-14526)
	* added support for FILS (IEEE 802.11ai) shared key authentication
	* added support for OWE (Opportunistic Wireless Encryption, RFC 8110;
	  and transition mode defined by WFA)
	* added support for DPP (Wi-Fi Device Provisioning Protocol)
	* added support for RSA 3k key case with Suite B 192-bit level
	* fixed Suite B PMKSA caching not to update PMKID during each 4-way
	  handshake
	* fixed EAP-pwd pre-processing with PasswordHashHash
	* added EAP-pwd client support for salted passwords
	* fixed a regression in TDLS prohibited bit validation
	* started to use estimated throughput to avoid undesired signal
	  strength based roaming decision
	* MACsec/MKA:
	  - new macsec_linux driver interface support for the Linux
	    kernel macsec module
	  - number of fixes and extensions
	* added support for external persistent storage of PMKSA cache
	  (PMKSA_GET/PMKSA_ADD control interface commands; and
	   MESH_PMKSA_GET/MESH_PMKSA_SET for the mesh case)
	* fixed mesh channel configuration pri/sec switch case
	* added support for beacon report
	* large number of other fixes, cleanup, and extensions
	* added support for randomizing local address for GAS queries
	  (gas_rand_mac_addr parameter)
	* fixed EAP-SIM/AKA/AKA' ext auth cases within TLS tunnel
	* added option for using random WPS UUID (auto_uuid=1)
	* added SHA256-hash support for OCSP certificate matching
	* fixed EAP-AKA' to add AT_KDF into Synchronization-Failure
	* fixed a regression in RSN pre-authentication candidate selection
	* added option to configure allowed group management cipher suites
	  (group_mgmt network profile parameter)
	* removed all PeerKey functionality
	* fixed nl80211 AP and mesh mode configuration regression with
	  Linux 4.15 and newer
	* added ap_isolate configuration option for AP mode
	* added support for nl80211 to offload 4-way handshake into the driver
	* added support for using wolfSSL cryptographic library
	* SAE
	  - added support for configuring SAE password separately of the
	    WPA2 PSK/passphrase
	  - fixed PTK and EAPOL-Key integrity and key-wrap algorithm selection
	    for SAE;
	    note: this is not backwards compatible, i.e., both the AP and
	    station side implementations will need to be update at the same
	    time to maintain interoperability
	  - added support for Password Identifier
	  - fixed FT-SAE PMKID matching
	* Hotspot 2.0
	  - added support for fetching of Operator Icon Metadata ANQP-element
	  - added support for Roaming Consortium Selection element
	  - added support for Terms and Conditions
	  - added support for OSEN connection in a shared RSN BSS
	  - added support for fetching Venue URL information
	* added support for using OpenSSL 1.1.1
	* FT
	  - disabled PMKSA caching with FT since it is not fully functional
	  - added support for SHA384 based AKM
	  - added support for BIP ciphers BIP-CMAC-256, BIP-GMAC-128,
	    BIP-GMAC-256 in addition to previously supported BIP-CMAC-128
	  - fixed additional IE inclusion in Reassociation Request frame when
	    using FT protocol
```

**Reply by alienatedsec** [reply]
It looks like I need to upgrade my dev rig as the latest `wpa_supplicant` has many dependencies not supported by `Debian 9`

**Reply by alienatedsec** [reply]
@Zerreth I managed the end goal and wondering how to test it now - Also, I may need to remove some modules to make the new `wpa_supplicant` and its libraries smaller

```
/tmp/sd/yi-hack-v5/wpa_210 # ./wpa_supplicant
Successfully initialized wpa_supplicant
wpa_supplicant v2.10
Copyright (c) 2003-2022, Jouni Malinen <j@w1.fi> and contributors

This software may be distributed under the terms of the BSD license.
See README for more details.

This product includes software developed by the OpenSSL Project
for use in the OpenSSL Toolkit (http://www.openssl.org/)

usage:
  wpa_supplicant [-BddhKLqqstuvW] [-P<pid file>] [-g<global ctrl>] \
        [-G<group>] \
        -i<ifname> -c<config file> [-C<ctrl>] [-D<driver>] [-p<driver_param>] \
        [-b<br_ifname>] [-e<entropy file>] [-f<debug file>] \
        [-o<override driver>] [-O<override ctrl>] \
        [-N -i<ifname> -c<conf> [-C<ctrl>] [-D<driver>] \
        [-m<P2P Device config file>] \
        [-p<driver_param>] [-b<br_ifname>] [-I<config file>] ...]

drivers:
  nl80211 = Linux nl80211/cfg80211
  wext = Linux wireless extensions (generic)
  wired = Wired Ethernet driver
options:
  -b = optional bridge interface name
  -B = run daemon in the background
  -c = Configuration file
  -C = ctrl_interface parameter (only used if -c is not)
  -d = increase debugging verbosity (-dd even more)
  -D = driver name (can be multiple drivers: nl80211,wext)
  -e = entropy file
  -f = log output to debug file instead of stdout
  -g = global ctrl_interface
  -G = global ctrl_interface group
  -h = show this help text
  -i = interface name
  -I = additional configuration file
  -K = include keys (passwords, etc.) in debug output
  -L = show license (BSD)
  -m = Configuration file for the P2P Device interface
  -N = start describing new interface
  -o = override driver parameter for new interfaces
  -O = override ctrl_interface parameter for new interfaces
  -p = driver parameters
  -P = PID file
  -q = decrease debugging verbosity (-qq even less)
  -s = log output to syslog instead of stdout
  -t = include timestamp in debug messages
  -u = enable DBus control interface
  -v = show version
  -W = wait for a control interface monitor before starting
example:
  wpa_supplicant -Dnl80211 -iwlan0 -c/etc/wpa_supplicant.conf
```

Here is the original which only has `wext` driver integrated.

```
/tmp/sd/yi-hack-v5/wpa_210 # wpa_supplicant
wpa_supplicant v2.6
Copyright (c) 2003-2016, Jouni Malinen <j@w1.fi> and contributors

This software may be distributed under the terms of the BSD license.
See README for more details.

usage:
  wpa_supplicant [-BddhKLqqtvW] [-P<pid file>] [-g<global ctrl>] \
        [-G<group>] \
        -i<ifname> -c<config file> [-C<ctrl>] [-D<driver>] [-p<driver_param>] \
        [-b<br_ifname>] [-e<entropy file>] \
        [-o<override driver>] [-O<override ctrl>] \
        [-N -i<ifname> -c<conf> [-C<ctrl>] [-D<driver>] \
        [-p<driver_param>] [-b<br_ifname>] [-I<config file>] ...]

drivers:
  wext = Linux wireless extensions (generic)
```

Also, some of the options are heavily expanded.

**Reply by InstantCassette** [reply]
If you need something tested, let me know.

**Reply by alienatedsec** [reply]
@Zerreth I will do that. I now have a version, which is half-size of my original build. It's important as libraries are dynamic and also take up space. I'll have a go over the weekend and may create another prerelease if I find it working well.

**Reply by alienatedsec** [reply]
@Zerreth A new pre-release with the latest `wpa_supplicant` 

https://github.com/alienatedsec/yi-hack-v5/releases/tag/0.4.1c

**Reply by InstantCassette** [reply]
@alienatedsec nice, will flash all 4 cameras tonight and report back anything is extra good or extra bad :)

**Reply by InstantCassette** [reply]
@alienatedsec been running all 4 cameras on 0.4.1c for a week **without issues**. 🎉 Even moved a few and they reconnected without problems. They're on the same network with 2 access points (Unifi-6-Enterprise) running WPA2/WPA3. I also noticed that disabling the snapshot & enabling the time OSD made the streams 100% stable, whereas without those it would freeze up now and then.

**Reply by alienatedsec** [reply]
@Zerreth promising :) so far mine had no issues too

**Reply by InstantCassette** [reply]
@alienatedsec Hate to say it but last saturday (so a week ago) 2 of the 4 cameras started bootlooping again at the exact same moment. And just minutes ago the 3rd camera has also started boot looping again. 😭

**Reply by alienatedsec** [reply]
Try commenting out the watchdog service from the system.sh and see if it boots correctly.

**Reply by InstantCassette** [reply]
> Try commenting out the watchdog service from the system.sh and see if it boots correctly.

What watchdog service ? I'm not finding anything marked as such in system.sh (0.4.1c)
Are you referring to this section ?
```shell
# Upgrade wpa_supplicant modules
ifconfig wlan0 up
if [ -f $YI_HACK_PREFIX/wpa/wpa_supplicant ]; then
    echo "---backing up wpa---"
    cp /home/base/tools/wpa_supplicant $YI_HACK_PREFIX/wpa/wpa_supplicant_backup
    cp /home/base/tools/wpa_cli $YI_HACK_PREFIX/wpa/wpa_cli_backup
    cp /home/base/tools/wpa_passphrase $YI_HACK_PREFIX/wpa/wpa_passphrase_backup
    killall watch_process
    killall wpa*
    mv /tmp/sd/yi-hack-v5/wpa/*.so* /home/lib/
    mv /tmp/sd/yi-hack-v5/wpa/wpa_supplicant /home/base/tools/
    mv /tmp/sd/yi-hack-v5/wpa/wpa_cli /home/base/tools/
    mv /tmp/sd/yi-hack-v5/wpa/wpa_passphrase /home/base/tools/
    reboot
    echo "---wpa upgrade done---"
else
	echo "---no wpa upgrade---"
fi
```

**Reply by alienatedsec** [reply]
Comment this line starting with `#` and start the camera

https://github.com/alienatedsec/yi-hack-v5/blob/4b7da2832ea50006ab313e9a2a90ca0a154bf501/src/static/static/home/yi-hack-v5/script/system.sh#L211

**Reply by InstantCassette** [reply]
No joy, now it stays orange for a while, then blinks blue and goes back to solid orange. (0.4.1c) Rinse and repeat. :)

**Reply by alienatedsec** [reply]
Logs will be much more useful if `./watch_process` is not doing anything

**Reply by InstantCassette** [reply]
wd_rtsp.log only contains a ton of:
```
2023-11-22 03:59:04 - Starting RTSP watchdog...
2023-11-22 03:59:02 - Starting RTSP watchdog...
```
also, my settings are: `DISABLE_CLOUD=yes` So it won't ever hit that line.
I also set `DISABLE_CLOUD=no` but nothing changed, still bootlooping, wd_rtsp.log only contains `Starting RTSP watchdog...`

**Comment by Porkenstien** [comment]
After reflashing the baseline, the camera is recording and connected to my network. However, the Web Interface is offline, SSH and FTP are also offline. The camera is responding to pings. Pulling the card shows new recordings in the folder.

Is the issue with the boot loop that the wifi/networking needs to wait for other services to start? Could it be a timing issue?

**Comment by cjd6568358** [comment]
0.3.8 doesn't work , 0.3.7 is work

**Reply by salado912** [reply]
had a camera (1 of 6) just stop working and could not get to the Web Interface , SSH and FTP. The camera did responding to pings (works in the  YI APP) so I manually tried flashing all the newer updates over and over and nothing worked, they would either boot loop or sometimes it just didnt work.
saw this comment that 0.3.7 works AND YES it worked again. not updating this camera for a while.
i have other cameras running V5 with latest updates fine, even some same versions (27US). dunno
